api/v1alpha1/dnsrecord_types.go

```diff
@@ -182,6 +101,37 @@ const (
 	DefaultGeo string = "default"
 )
 
+// GetRootDomain returns the shortest domain that is shared across all spec.Endpoints dns names.
+// Validates that all endpoints share an equal root domain and returns an error if they don't.
+func (s *DNSRecord) GetRootDomain() (string, error) {
+	domain := ""
+	dnsNames := []string{}
+	for idx := range s.Spec.Endpoints {
+		dnsNames = append(dnsNames, s.Spec.Endpoints[idx].DNSName)
+	}
+	for idx := range dnsNames {
+		if domain == "" {
+			domain = dnsNames[0]
```

Comment:
should this be `domain = dnsNames[idx]`?

Reply:
Yeah,  can probably just be:

```.go
for idx := range dnsNames {
	if domain == "" || len(domain) > len(dnsNames[idx]) {
		domain = dnsNames[idx]
	}
}
```

Will update.